File: Examples/android/CMeSdkJava/app_rework/src/main/java/me/connect/sdk/java/sample/connections/ConnectionsFragment.java
package me.connect.sdk.java.sample.connections;

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.google.zxing.integration.android.IntentIntegrator;
import com.google.zxing.integration.android.IntentResult;

import me.connect.sdk.java.sample.databinding.ConnectionsFragmentBinding;

public class ConnectionsFragment extends Fragment {

    private ConnectionsFragmentBinding binding;
    private ConnectionsViewModel model;

    public static ConnectionsFragment newInstance() {
        return new ConnectionsFragment();
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        binding = ConnectionsFragmentBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(getContext());
        binding.connectionList.setLayoutManager(layoutManager);
        ConnectionsAdapter adapter = new ConnectionsAdapter();
        binding.connectionList.setAdapter(adapter);

        model = new ViewModelProvider(requireActivity()).get(ConnectionsViewModel.class);
        model.getConnections().observe(getViewLifecycleOwner(), adapter::setData);

        binding.buttonAddConnection.setOnClickListener(v -> {
            String invite = binding.editTextConnection.getText().toString();
            performNewConnection(invite);
        });

        binding.buttonQr.setOnClickListener(v -> {
            IntentIntegrator
                    .forSupportFragment(this)
                    .setPrompt("Scan invitation QR code")
                    .setOrientationLocked(true)
                    .initiateScan();
        });
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        IntentResult result = IntentIntegrator.parseActivityResult(requestCode, resultCode, data);
        if (result != null) {
            if (result.getContents() != null) {
                String invite = result.getContents();
                binding.editTextConnection.setText(invite);
                performNewConnection(invite);
            }
        } else {
            super.onActivityResult(requestCode, resultCode, data);
        }
    }

    private void performNewConnection(String invite) {
        binding.buttonAddConnection.setEnabled(false);
        binding.buttonQr.setEnabled(false);
        model.newConnection(invite).observeOnce(getViewLifecycleOwner(), status -> {
            Toast.makeText(getActivity(), "New connection processed", Toast.LENGTH_SHORT).show();
            if (status) {
                binding.editTextConnection.setText(null);
            }
            binding.buttonAddConnection.setEnabled(true);
            binding.buttonQr.setEnabled(true);
        });
    }
}

File: Examples/android/CMeSdkJava/app_rework_kt/src/main/kotlin/me/connect/sdk/java/samplekt/connections/ConnectionsFragment.kt
package me.connect.sdk.java.samplekt.connections

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import androidx.lifecycle.Observer
import androidx.recyclerview.widget.LinearLayoutManager
import com.google.zxing.integration.android.IntentIntegrator
import me.connect.sdk.java.samplekt.db.entity.Connection
import me.connect.sdk.java.samplekt.databinding.ConnectionsFragmentBinding


class ConnectionsFragment : Fragment() {
    private lateinit var binding: ConnectionsFragmentBinding
    private val model: ConnectionsViewModel by viewModels()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                               savedInstanceState: Bundle?): View {
        binding = ConnectionsFragmentBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val layoutManager = LinearLayoutManager(context)
        val adapter = ConnectionsAdapter()
        model.getConnections().observe(viewLifecycleOwner, Observer<List<Connection>> { adapter.setData(it) })
        binding.connectionList.layoutManager = layoutManager
        binding.connectionList.adapter = adapter
        binding.buttonAddConnection.setOnClickListener {
            val invite: String = binding.editTextConnection.text.toString()
            performNewConnection(invite)
        }
        binding.buttonQr.setOnClickListener {
            IntentIntegrator
                    .forSupportFragment(this)
                    .setPrompt("Scan invitation QR code")
                    .setOrientationLocked(true)
                    .initiateScan()
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        val result = IntentIntegrator.parseActivityResult(requestCode, resultCode, data)
        if (result != null) {
            if (result.contents != null) {
                val invite = result.contents
                binding.editTextConnection.setText(invite)
                performNewConnection(invite)
            }
        } else {
            super.onActivityResult(requestCode, resultCode, data)
        }
    }

    private fun performNewConnection(invite: String) {
        binding.buttonAddConnection.isEnabled = false
        binding.buttonQr.isEnabled = false
        model.newConnection(invite).observeOnce(viewLifecycleOwner, Observer { status: Boolean ->
            Toast.makeText(activity, "New connection processed", Toast.LENGTH_SHORT).show()
            if (status) {
                binding.editTextConnection.text = null
            }
            binding.buttonAddConnection.isEnabled = true
            binding.buttonQr.isEnabled = true
        })
    }

    companion object {
        fun newInstance(): ConnectionsFragment = ConnectionsFragment()
    }
}